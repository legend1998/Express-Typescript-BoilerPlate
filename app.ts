import express from "express";
import cookieParser from "cookie-parser";
import logger from "morgan";
import { configDotenv } from "dotenv";
import indexRouter from "./routes/index";
import userRouter from "./routes/users";
import favoriteRouter from "./routes/favorite";
import { errorHandler } from "./src/exception/Errorhandlers";
import connectDB from "./src/database/mongodb";

configDotenv();
const app = express();

// connect mongodb
connectDB();

app.use(logger("dev"));
app.use(express.json());
app.use(express.urlencoded({ extended: false }));
app.use(cookieParser());

// app.use(express.static(path.join(__dirname, "public")));

app.use("/", indexRouter);

// global error handler
app.use(errorHandler);

export default app;
