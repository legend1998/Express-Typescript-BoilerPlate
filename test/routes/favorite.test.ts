import request from "supertest";
import mongoose from "mongoose";
import app from "../../app";

const userId = "666dbcd805052a0673aeddcb";
const movieId = "666dbc47d973977242b614c8";
const contentType = "Movie";

// beforeAll(async () => {
//   await mongoose.connect(
//     "mongodb+srv://suman:xFtw06WwOPMq8Ppd@cluster0.09rmyvy.mongodb.net/stage"
//   );
// });

// unit testing

describe("GET /favorite/fetch/asdf", () => {
  it(`should return 400 if userId parameter is invalid`, async () => {
    const res = await request(app)
      .get("/favorite/fetch/asdf")
      .set("Authorization", "suman mypass");

    expect(res.statusCode).toBe(400);
    expect(res.body).toMatchSnapshot({ error: expect.any(String) });
  });

  it(`should return mylist for ther user ${userId} `, async () => {
    const res = await request(app)
      .get("/favorite/fetch/" + userId)
      .set("Authorization", "suman mypass");
    expect(res.statusCode).toBe(200);
    expect(res.body.length).toBeGreaterThan(2);
  });
});

describe("PUT /favorite/update", () => {
  it(`should return 400 if userId query parameter is missing`, async () => {
    const res = await request(app)
      .put("/favorite/update")
      .set("Authorization", "suman mypass")
      .query({ userId: userId });

    expect(res.statusCode).toBe(400);
    expect(res.body).toMatchSnapshot({ error: expect.any(String) });
  });

  it(`should return 400 if contentId query parameter is missing`, async () => {
    const res = await request(app)
      .put("/favorite/update")
      .set("Authorization", "suman mypass")
      .query({ userId: userId, type: contentType });

    expect(res.statusCode).toBe(400);
    expect(res.body).toMatchSnapshot({ error: expect.any(String) });
  });

  it(`should return 400 if type query parameter is missing`, async () => {
    const res = await request(app)
      .put("/favorite/update")
      .set("Authorization", "suman mypass")
      .query({ userId: userId, contentId: movieId });

    expect(res.statusCode).toBe(400);
    expect(res.body).toMatchSnapshot({ error: expect.any(String) });
  });

  it(`should return 200 with objectId add this item - Movie, id-  ${movieId} ,userId - ${userId}`, async () => {
    const res = await request(app)
      .put("/favorite/update")
      .set("Authorization", "suman mypass")
      .query({ userId: userId, contentId: movieId, type: contentType });
    expect(res.statusCode).toBe(200);
    expect(res.body).toMatchSnapshot({
      id: expect.any(String),
      exists: expect.any(Boolean),
    });
  });
});

describe("DELETE /favorite ", () => {
  it(`should return 400 if id query parameter is missing`, async () => {
    const res = await request(app)
      .put("/favorite/delete")
      .set("Authorization", "suman mypass")
      .query({ userId: userId });
    expect(res.statusCode).toBe(400);
    expect(res.body).toMatchSnapshot({ error: expect.any(String) });
  });

  it(`should return 200 for userid ${userId} & itemId ${movieId}`, async () => {
    const res = await request(app)
      .delete("/favorite/delete")
      .set("Authorization", "suman mypass")
      .query({ id: movieId, userId: userId });

    expect(res.statusCode).toBe(200);
    expect(res.body).toMatchSnapshot({ deleted: expect.any(Number) });
  });
});
