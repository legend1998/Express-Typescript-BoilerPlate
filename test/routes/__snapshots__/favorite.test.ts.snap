// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DELETE /favorite  should return 200 for userid 666dbcd805052a0673aeddcb & itemId 666dbc47d973977242b614c8 1`] = `
{
  "deleted": Any<Number>,
}
`;

exports[`GET /favorite/fetch/asdf should return 400 if userId parameter is invalid 1`] = `
{
  "error": Any<String>,
}
`;

exports[`PUT /favorite/update should return 200 with objectId add this item - Movie, id-  666dbc47d973977242b614c8 ,userId - 666dbcd805052a0673aeddcb 1`] = `
{
  "exists": Any<Boolean>,
  "id": Any<String>,
}
`;

exports[`PUT /favorite/update should return 400 if contentId query parameter is missing 1`] = `
{
  "error": Any<String>,
}
`;

exports[`PUT /favorite/update should return 400 if type query parameter is missing 1`] = `
{
  "error": Any<String>,
}
`;

exports[`PUT /favorite/update should return 400 if userId query parameter is missing 1`] = `
{
  "error": Any<String>,
}
`;
